shared/src/address/mod.rs

```diff
@@ -266,6 +276,29 @@ impl FromStr for Address {
             }
         };
 
+        fn validate_and_split_checksum<'a>(
+            protocol: Protocol,
+            prefix: Option<&[u8]>,
+            payload: &'a [u8],
+        ) -> Result<&'a [u8], Error> {
+            if payload.len() < CHECKSUM_HASH_LEN {
```

Comment:
<= ?

Reply:
Technically the payload can be empty.

Reply:
Or, at least, that's not what we're asserting here.